webapp-django/crashstats/manage/admin.py

```diff
@@ -271,31 +273,57 @@ def buildhub_check(request):
     session = session_with_retries(buildhub_api)
 
     for pv in pv_data:
-        query = {
-            'source.product': pv['product'].lower(),
-            'build.id': '"%s"' % pv['build_id'],
-            'target.channel': pv['channel']
+        # Lookup (product, buildid, channel) and exclude versions with "rc" in
+        # them because we're looking at the beta channel
+        es_query = {
+            'query': {
+                'bool': {
+                    'must': {
+                        'match_all': {}
+                    },
+                    'filter': [
+                        {
+                            'term': {
+                                'source.product': pv['product'].lower()
+                            }
+                        },
+                        {
+                            'term': {
+                                'build.id': str(pv['build_id'])
+                            }
+                        },
+                        {
+                            'term': {
+                                'target.channel': pv['channel']
+                            }
+                        },
+                    ],
+                    'must_not': {
+                        'wildcard': {
+                            'target.version': '*rc*'
+                        }
+                    }
+                }
+            },
+            'size': 1
         }
+
         buildhub_start = time.time()
-        resp = session.get(buildhub_api, params=query)
+        resp = session.post(buildhub_api, data=json.dumps(es_query))
         buildhub_time = buildhub_time + (time.time() - buildhub_start)
```

Comment:
This change drops this page from taking 50+ seconds to run (of which all of that is taken up by querying Buildhub) to taking 1 second to run. It's pretty incredible.